src/Contract/Translator.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace ipl\Stdlib;
+
+/**
+ * Representation of translators
+ */
+interface Translator
+{
+    /**
+     * Translate a message
+     *
+     * @param string $message
+     * @param string $context Message context
+     *
+     * @return string Translated message or original message if no translation found
+     */
+    public function translate($message, $context = null);
+
+    /**
+     * Translate a message in the given domain
+     *
+     * If no translation is found in the specified domain, the translation is also searched for in the default domain.
+     *
+     * @param string $domain
+     * @param string $message
+     * @param string $context Message context
+     *
+     * @return string Translated message or original message if no translation found
+     */
+    public function translateInDomain($domain, $message, $context = null);
+
+    /**
+     * Translate a plural message
+     *
+     * If no translation is found, the returned message is based on the given number to decide between the original
+     * singular and plural messages.
```

Comment:
That's not the case if a translation is found? It's not, but that's how I read this.